src/condor_starter.V6.1/condor_diagnostic_send_ep_logs.cpp

```diff
@@ -0,0 +1,165 @@
+#include "condor_common.h"
+#include "condor_config.h"
+
+#include "shortfile.h"
+#include "safe_open.h"
+
+#include <filesystem>
+#include <functional>
+#include <fstream>
+
+bool
+dumpFileToStandardOut(
+  const std::filesystem::path & path,
+  std::function<bool(const char *)> filter
+) {
+    std::ifstream i( path );
+    if(! i.good()) {
+        fprintf( stderr, "Failed to open %s\n", path.string().c_str() );
+        return false;
+    }
+
+    std::string line;
+    while( std::getline( i, line ).good() ) {
+        if( filter( line.c_str() ) ) {
+            fprintf( stdout, "%s\n", line.c_str() );
+        }
+    }
+    if(! i.eof()) { return false; }
+
+    return true;
+}
+
+
+// We can't just check for `pattern`, because that will miss lines from the
+// FTO if it's a forked child.  Instead, we look for the first banner line;
+// if it also contains `pattern`, we start including log lines.  We stop
+// including log lines when we find a first banner line that doesn't
+// contain `pattern` (note that we can't check for this _before_ we find
+// our own first banner line, because there are other starters in the log).
+//
+// As this function is currently called, we'll never see the next starter's
+// banner (because this starter hasn't exited yet).
+//
+bool
+relevantPartOfLog( const std::string & pattern, const char * line ) {
+    static bool saw_our_banner_line = false;
+    static bool saw_bad_banner_line = false;
+
+    if( NULL != strstr( line, "******************************************************" )) {
+        bool good_pid = NULL != strstr( line, pattern.c_str() );
+
+        if(! saw_our_banner_line) {
+            saw_our_banner_line = good_pid;
+        } else {
+            saw_bad_banner_line = (! good_pid);
+        }
+    }
+
+    if( saw_our_banner_line && ! saw_bad_banner_line ) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+bool
+printGlobalStarterLog( pid_t pid ) {
```

Comment:
Do we also want to send home the `StarterLog.testing` ?

Reply:
That's presently out of scope, and while I guess we could, I hadn't thought it would be very useful.  I could make it a feature request ticket, if you like.